mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -1077,58 +1201,53 @@ private class LogisticCostFun(
     fitIntercept: Boolean,
     standardization: Boolean,
     featuresStd: Array[Double],
-    featuresMean: Array[Double],
-    regParamL2: Double) extends DiffFunction[BDV[Double]] {
+    regParamL2: Double,
+    multinomial: Boolean,
+    standardize: Boolean) extends DiffFunction[BDV[Double]] {
 
   override def calculate(coefficients: BDV[Double]): (Double, BDV[Double]) = {
-    val numFeatures = featuresStd.length
     val coeffs = Vectors.fromBreeze(coefficients)
-    val n = coeffs.size
+    val bcCoeffs = instances.context.broadcast(coeffs)
```

Comment:
I noticed some (somewhat modest) performance gains when explicitly broadcasting the coefficients when the number of coefficients was large.